api/models/comments-models.js

```diff
@@ -2,24 +2,25 @@ const db = require("../../db/connection");
 const format = require("pg-format");
 
 exports.insertCommentByArticleId = (article_id, { username, body }) => {
-  const sqlStatement = format(
-    `
-  INSERT INTO comments
-    (article_id, author, body)
-  VALUES
-    (%s, %L, %L)
-  RETURNING *;`,
-    article_id,
-    username,
-    body
-  );
+  if (!username || !body) {
+    return Promise.reject({ status: 400, msg: "bad request" });
+  }
 
-  return db.query(sqlStatement).then((body) => {
-    if (!body.rows.length) {
-      return Promise.reject({ status: 404, msg: "not found" });
-    }
-    return body.rows[0];
-  });
+  return db
+    .query(
+      `INSERT INTO comments
+          (article_id, author, body)
+        VALUES
+          ($1, $2, $3)
+        RETURNING *;`,
+      [article_id, username, body]
+    )
+    .then((body) => {
+      if (!body.rows.length) {
+        return Promise.reject({ status: 404, msg: "not found" });
+      }
+      return body.rows[0];
+    });
```

Comment:
Nice changes to this model to avoid injection concerns.
You may want to differentiate your error messages - You are returning a message of `not found` for both an article that doesn't exist and a user that doesn't exist. When you get to front end it could be useful to know which one is which!